src/System.Management.Automation/commands/utility/FormatAndOutput/out-console/OutConsole.cs

```diff
@@ -102,6 +98,7 @@ protected override void ProcessRecord()
         {
             if (Transcript)
             {
+                Host.UI.TranscribeOnly = true;
```

Comment:
Just asking - but should TranscribeOnly be a counter that is incremented each time it's true, and decremented in Dispose?

I'm not sure this is the correct semantic, but it seems right - like it shouldn't be possible to turn it off as long as somebody wants it on.